File: jobs/Mobile/packages/tmdb_api/lib/src/models/movie.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:tmdb_api/src/models/models.dart';
import 'package:tmdb_api/src/models/movie_base.dart';
import 'package:tmdb_api/src/utils/tmdb_utils.dart';

part 'movie.g.dart';

@JsonSerializable()
class Movie extends MovieBase {
  final int budget;
  final List<Genre> genres;
  final String? homepage;
  final String? imdbId;
  final String? overview;
  final List<Company> productionCompanies;
  final List<Country> productionCountries;
  final int revenue;
  final int? runtime;
  final List<Language> spokenLanguages;
  final String status;
  final String? tagline;

  Movie({
    required adult,
    backdropPath,
    required id,
    required originalLanguage,
    required originalTitle,
    required popularity,
    required posterPath,
    releaseDate,
    required title,
    required video,
    required voteAverage,
    required voteCount,
    required this.budget,
    required this.genres,
    required this.homepage,
    required this.imdbId,
    required this.overview,
    required this.productionCompanies,
    required this.productionCountries,
    required this.revenue,
    required this.runtime,
    required this.spokenLanguages,
    required this.status,
    required this.tagline,
  }) : super(
            adult: adult,
            backdropPath: backdropPath,
            id: id,
            originalLanguage: originalLanguage,
            originalTitle: originalTitle,
            popularity: popularity,
            posterPath: posterPath,
            releaseDate: releaseDate,
            title: title,
            video: video,
            voteAverage: voteAverage,
            voteCount: voteCount);

  factory Movie.fromJson(Map<String, dynamic> json) => _$MovieFromJson(json);

  Map<String, dynamic> toJson() => _$MovieToJson(this);
}

File: jobs/Frontend/movies/src/models/movie.ts
import { CountryCode } from "./country";
import { DateString } from "./dateString";
import { LanguageCode } from "./language";

export interface Movie {
  adult: boolean;
  backdrop_path: string | null;
  genre_ids: number[];
  id: number;
  original_language: LanguageCode;
  original_title: string;
  overview: string;
  popularity: number;
  poster_path: string | null;
  release_date: DateString;
  title: string;
  video: boolean;
  vote_average: number;
  vote_count: number;
}

export interface MovieResponse {
  page: number;
  results: Movie[];
  total_pages: number;
  total_results: number;
}

interface Collection {
  id: number;
  name: string;
  poster_path: string;
  backdrop_path: string;
}

interface Genre {
  id: number;
  name: string;
}

export interface MovieDetails {
  adult: boolean;
  backdrop_path: Collection | null;
  belongs_to_collection: string | null;
  budget: number;
  genres: Genre[];
  homepage: string;
  id: number;
  imdb_id: string | null;
  origin_country: CountryCode[];
  original_language: LanguageCode;
  original_title: string;
  overview: string;
  popularity: number;
  poster_path: string | null;
  production_companies: Array<{
    id: number;
    logo_path: string | null;
    name: string;
    origin_country: CountryCode;
  }>;
  production_countries: Array<{
    iso_3166_1: CountryCode;
    name: string;
  }>;
  release_date: DateString;
  revenue: number;
  runtime: number;
  spoken_languages: Array<{
    iso_639_1: LanguageCode;
    name: string;
  }>;
  status: string;
  tagline: string;
  title: string;
  video: boolean;
  vote_average: number;
  vote_count: number;
}
